### ISSUE TITLE
Incomplete webextension listed in search results page

### ISSUE THREAD

**Issue by ValentinaPC** [mozilla/addons-server#5588]
Steps to reproduce:
1. Go to a webextension details page an click on the `firefox57` tag option i.e. https://addons.allizom.org/en-US/firefox/addon/tampermonkey-%E5%9F%BC%E5%9F%BC-%C3%A3-%C4%83%C4%83/?src=search
2. Observe first webextension from the search results page (WebExt-simple-stage)

Expected results:
Incomplete webextensions are not listed in the search results.

Actual results:
Incomplete webextensions are not listed in the search results.

Notes/Issues:
- clicking on the webextension will refresh the page
- the webextension has 3 disabled versions - rejected using the multiple file rejection option
- not being able to reproduce with other webext 
- also the webextensions compatibility label is not listed

Verified on AMO-stage FF53(Win 7).
Video and screenshot for this issue:
![dev_hub](https://user-images.githubusercontent.com/15685960/26882959-8dc8e13a-4ba4-11e7-94c6-9969075f6227.png)
![incomp](https://user-images.githubusercontent.com/15685960/26882964-91d142ae-4ba4-11e7-9db6-e426d619d948.gif)

**Comment by diox** [comment]
> the webextension has 3 disabled versions - rejected using the multiple file rejection option

So, that looks like it's an issue with that specific action then.

**Comment by diox** [comment]
@ValentinaPC can you reproduce the problem on dev using a newly created add-on? I need STRs from a fresh add-on.

**Comment by ValentinaPC** [comment]
@diox : Unfortunately I was not able to reproduce for other webextension as I mentioned on notes, but I'll follow up on this try to add some STRs.

**Comment by diox** [comment]
@ValentinaPC if you haven't already, try playing around with the reject multiple versions action, since I suspect that's the root cause here.

**Comment by ValentinaPC** [comment]
@diox : seems like the issue **_is taking effect ONLY after a while_**, because now, looking at https://addons.allizom.org/en-US/firefox/tag/firefox57 instead of only one (first) webextension with problems, there are a bunch more....and they are all webextension with all versions rejected using multiple-rejection a few days ago while logging this.

STRs:
Pick an approved extension and reject all its versions using Reject Multiple Versions option from Reviewer Tools.
Example: "HTTP Resource Test" from https://addons.allizom.org/en-US/firefox/search/?q=test&appver=53.0&platform=windows

Due to the fact that this issue has clear STR and is reproducing every time, I'll add a more severe label.
Thanks!

**Comment by muffinresearch** [comment]
Following @diox's run through of this issue in today's standup downgrading to p2.

**Comment by diox** [comment]
I think this is due to the reindex taking place too early. The proper fix would be something like [post-request-task](https://github.com/mozilla/django-post-request-task), which we never enabled because it was causing various issues like #1552

**Comment by diox** [comment]
Closing in favor of https://github.com/mozilla/addons-server/issues/5999
